File: 9.Custom_Sort/Javascript/13.UseSideLane.js
function canSortTrucks(trucks, n) {
    const sideLane = [];
    let current = 1;

    for (let i = 0; i < trucks.length; i++) {
        const truck = trucks[i];

        while (sideLane.length > 0 && sideLane[sideLane.length - 1] === current) {
            sideLane.pop();
            current++;
        }

        if (truck === current) {
            current++;
        } else {
            sideLane.push(truck);
        }
    }

    while (sideLane.length > 0 && sideLane[sideLane.length - 1] === current) {
        sideLane.pop();
        current++;  
    }

    return current === n + 1 ? "yes" : "no";
}

const n = 5;
const trucks = [5, 1, 2, 4, 3];
const res = canSortTrucks(trucks, n);
console.log(res);

File: 9.Custom_Sort/Python/13.UseSideLane.py
def can_sort_trucks(trucks,n):
    side_lane = []
    current = 1
    
    for truck in trucks:
        while side_lane and side_lane[-1] == current:
            side_lane.pop()
            current += 1
            
        if truck == current:
            current += 1
        else:
            side_lane.append(truck)
    
    while side_lane and side_lane[-1] == current:
        side_lane.pop()
        current += 1
    
    return "yes" if current == n + 1 else "no"

n = 5
trucks = [5,1,2,4,3]
res = can_sort_trucks(trucks,n)
print(res)